enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.ts

```diff
@@ -67,28 +73,57 @@ export const useInitData = ({
   useEffect(() => {
     api.basename = config.metabaseInstanceUrl;
 
-    if (config.authType === "jwt") {
+    if (config.authType === "jwt" && config.jwtProviderUri) {
       api.onBeforeRequest = () =>
         dispatch(getOrRefreshSession(config.jwtProviderUri));
       api.sessionToken = sessionTokenState?.token?.id;
+      dispatch(setLoginStatus({ status: "initialized" }));
     } else if (config.authType === "apiKey" && config.apiKey) {
       api.apiKey = config.apiKey;
+      dispatch(setLoginStatus({ status: "initialized" }));
     } else {
-      dispatch(setIsLoggedIn(false));
-      return;
+      dispatch(
+        setLoginStatus({
+          status: "error",
+          error: new Error(getErrorMessage(config.authType)),
+        }),
+      );
     }
-
-    Promise.all([
-      dispatch(refreshCurrentUser()),
-      dispatch(reloadSettings()),
-    ]).then(() => {
-      dispatch(setIsInitialized(true));
-      dispatch(setIsLoggedIn(true));
-    });
   }, [config, dispatch, sessionTokenState]);
 
-  return {
-    isLoggedIn,
-    isInitialized,
-  };
+  useEffect(() => {
+    const fetchData = async () => {
+      if (loginStatus.status === "initialized") {
+        dispatch(setLoginStatus({ status: "loading" }));
+
+        try {
+          const [userData] = await Promise.all([
+            dispatch(refreshCurrentUser()),
+            dispatch(reloadSettings()),
+          ]);
+
+          if (userData.meta.requestStatus === "rejected") {
```

Comment:
`reloadSettings` doesn't return anything atm (`undefined`), so you need to return the result to read it.
https://github.com/metabase/metabase/blob/82b8f49274fafeb9347d3a0eaca812f798d1b084/frontend/src/metabase/admin/settings/settings.js#L19-L24

Once you return it you should be able to at least read the result from `refreshSiteSettings` (called within `reloadSettings`) since it's being handled by Redux toolkit (Redux toolkit adds this `meta` object)

![image](https://github.com/metabase/metabase/assets/1937582/805d4638-e45e-4f2f-b159-d028325d76df)

Then you should be able to utilize both user and site settings Response statuses